controller/space_template.go

```diff
@@ -0,0 +1,171 @@
+package controller
+
+import (
+	"encoding/base64"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/jsonapi"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/rest"
+	"github.com/almighty/almighty-core/spacetemplate"
+	"github.com/goadesign/goa"
+)
+
+const APIStringType = "spacetemplates"
+
+// SpaceTemplateController implements the space_template resource.
+type SpaceTemplateController struct {
+	*goa.Controller
+	db     application.DB
+	config SpaceTemplateControllerConfiguration
+}
+
+// SpaceTemplateControllerConfiguration the configuration for the SpaceTemplateController
+type SpaceTemplateControllerConfiguration interface {
+	GetCacheControlSpaceTemplates() string
+}
+
+// NewSpaceTemplateController creates a space_template controller.
+func NewSpaceTemplateController(service *goa.Service, db application.DB, config SpaceTemplateControllerConfiguration) *SpaceTemplateController {
+	return &SpaceTemplateController{
+		Controller: service.NewController("SpaceTemplateController"),
+		db:         db,
+		config:     config,
+	}
+}
+
+// Create runs the create action.
+func (c *SpaceTemplateController) Create(ctx *app.CreateSpaceTemplateContext) error {
+	var err error
+	defer func() {
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
```

Comment:
True and since you seem to dislike the defer approach, I've done it differently in e968b365bb7d31e37356f9ef9d46dbf7aca80329